reporting-engine-worker/src/main/java/org/ihtsdo/termserver/scripting/reports/MismatchedLaterality.java

```diff
@@ -3,199 +3,126 @@
-		//lateralized - but we're only expecting that in a body structure
-		for (Relationship r : c.getRelationships(CharacteristicType.INFERRED_RELATIONSHIP, ActiveState.ACTIVE)) {
-			if ((r.isNotConcrete()) && (r.getTarget().equals(LEFT) || r.getTarget().equals(RIGHT))) {
-				String semTag = SnomedUtils.deconstructFSN(c.getFsn())[1];
-				if (!semTag.equals("(body structure)") && !reportedSuspect.contains(c)) {
-					report(QUATERNARY_REPORT, c, c.toExpression(CharacteristicType.INFERRED_RELATIONSHIP));
-					reportedSuspect.add(c);
-				}
-				return true;
-			} else if (level == 0 && r.isNotConcrete() && hasLateralizedModel(r.getTarget(), 1)) {
-				return true;
-			}
-		}
-		return false;
-	}
-	
-	private boolean hasBilateralizedModel(Concept c, int level) throws TermServerScriptException {
-		//So either a target value is a type of laterality or it is itself 
-		//lateralized - but we're only expecting that in a body structure
-		//Both must be present
-		boolean leftPresent = false;
-		boolean rightPresent = false;
-		for (Relationship r : c.getRelationships(CharacteristicType.INFERRED_RELATIONSHIP, ActiveState.ACTIVE)) {
-			if (r.isConcrete()) {
-				continue;
-			}
-			if (r.getTarget().equals(LEFT)) {
-				String semTag = SnomedUtils.deconstructFSN(c.getFsn())[1];
-				if (!semTag.equals("(body structure)") && !reportedSuspect.contains(c)) {
-					report (TERTIARY_REPORT, c ,c.toExpression(CharacteristicType.INFERRED_RELATIONSHIP));
-					reportedSuspect.add(c);
-				}
-				leftPresent = true;
-			} if (r.getTarget().equals(RIGHT)) {
-				String semTag = SnomedUtils.deconstructFSN(c.getFsn())[1];
-				if (!semTag.equals("(body structure)") && !reportedSuspect.contains(c)) {
-					report (TERTIARY_REPORT, c ,c.toExpression(CharacteristicType.INFERRED_RELATIONSHIP));
-					reportedSuspect.add(c);
-				}
-				rightPresent = true;
-			} else if (level == 0 && hasLaterality(r.getTarget(), LEFT)) {
-				leftPresent = true;
-			} else if (level == 0 && hasLaterality(r.getTarget(), RIGHT)) {
-				rightPresent = true;
-			}
-		}
-		return leftPresent && rightPresent;
-	}
-
-	private boolean hasLaterality(Concept c, Concept laterality) {
-		for (Relationship r : c.getRelationships(CharacteristicType.INFERRED_RELATIONSHIP, ActiveState.ACTIVE)) {
-			if (r.isNotConcrete() && r.getTarget().equals(laterality)) {
-				return true;
-			}
-		}
-		return false;
-	}
+    private final Map<String, Concept> fsnMap = new HashMap<>();
+
+    public static void main(String[] args) throws TermServerScriptException, IOException {
+        Map<String, String> params = new HashMap<>();
+        params.put("legacy", "false"); // Toggle whether to process all Concepts or only those that are new/modified.
+        params.put("notYetMembers", "true"); // Toggle whether to process Concepts not yet in the lateralisable reference set.
+        params.put("alreadyMembers", "true"); // Toggle whether to process Concepts already in the lateralisable reference set.
```

Comment:
Some general flags I think would be useful. By default, only new/modified concepts will be inspected.